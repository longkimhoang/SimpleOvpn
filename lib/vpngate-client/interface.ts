import {IVpnServer} from './models';

export interface FetchServersOptions {
  signal?: AbortSignal;
}

export interface VpnServerRepository {
  vpnServers: readonly IVpnServer[];
  isFetching: boolean;
  fetchServers(options?: FetchServersOptions): void;
}
