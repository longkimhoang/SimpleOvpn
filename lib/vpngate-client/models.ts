import Realm, {ObjectSchema} from 'realm';

export interface IVpnServer {
  hostName: string;
  ipAddress: string;
  score: number;
  ping: number;
  speed: number;
  country: string;
  countryCode: string;
  operator: string;
  base64EncodedOvpnConfig: string;
}

export class VpnServer extends Realm.Object implements IVpnServer {
  _id!: Realm.BSON.ObjectId;
  hostName!: string;
  ipAddress!: string;
  score!: number;
  ping!: number;
  speed!: number;
  country!: string;
  countryCode!: string;
  operator!: string;
  base64EncodedOvpnConfig!: string;

  static generate(
    properties: IVpnServer,
  ): RealmInsertionModel<VpnServer> {
    return {
      ...properties,
      _id: new Realm.BSON.ObjectId(),
    };
  }

  static schema: ObjectSchema = {
    name: 'VpnServer',
    primaryKey: '_id',
    properties: {
      _id: 'objectId',
      hostName: 'string',
      ipAddress: 'string',
      score: {
        type: 'int',
        indexed: true,
      },
      ping: 'int',
      country: 'string',
      countryCode: 'string',
      operator: 'string',
      base64EncodedOvpnConfig: 'string',
    },
  };
}
