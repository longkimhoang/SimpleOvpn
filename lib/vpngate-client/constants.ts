export const VPN_GATE_SERVERS_LIST_URL = 'https://www.vpngate.net/api/iphone/';
