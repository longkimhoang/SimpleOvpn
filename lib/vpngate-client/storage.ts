import {useMMKV, useMMKVObject} from 'react-native-mmkv';
import {IVpnServer} from './models';

/**
 * A type that manages local storage of fetched VPN servers
 */
export interface VpnServerStorage {
  data: readonly IVpnServer[];
  write(data: readonly IVpnServer[]): void | Promise<void>;
}

export function useMmkvVpnServerStorage(): VpnServerStorage {
  const storage = useMMKV({
    id: 'vpngate',
  });

  const [servers, setServers] = useMMKVObject<readonly IVpnServer[]>(
    'vpn-servers',
    storage,
  );

  return {
    data: servers ?? [],
    write: setServers,
  };
}
