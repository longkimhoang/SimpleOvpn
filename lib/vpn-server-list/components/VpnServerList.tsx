import React from 'react';
import {ListRenderItemInfo} from 'react-native';
import {FlatList} from 'react-native-gesture-handler';
import {useTheme} from 'react-native-paper';
import {IVpnServer} from '../../vpngate-client/models';
import {VPN_SERVER_CELL_HEIGHT} from './constants';
import VpnServerCell from './VpnServerCell';

//#region FlatList implementations

function renderVpnServerItem({item}: ListRenderItemInfo<IVpnServer>) {
  return <VpnServerCell data={item} />;
}

function getVpnServerCellLayout(
  _data: IVpnServer[] | undefined | null,
  index: number,
) {
  return {
    length: VPN_SERVER_CELL_HEIGHT,
    offset: VPN_SERVER_CELL_HEIGHT * index,
    index,
  };
}

//#endregion

function extractKeyForVpnServerItem(item: IVpnServer): string {
  return item.ipAddress;
}

//#region Component

export interface VpnServerListProps {
  vpnServers: readonly IVpnServer[];
}

function VpnServerList({vpnServers}: VpnServerListProps) {
  const {colors} = useTheme();

  return (
    <FlatList
      data={vpnServers}
      renderItem={renderVpnServerItem}
      getItemLayout={getVpnServerCellLayout}
      keyExtractor={extractKeyForVpnServerItem}
      style={[
        {
          backgroundColor: colors.surface,
        },
      ]}
    />
  );
}

//#endregion

export default VpnServerList;
