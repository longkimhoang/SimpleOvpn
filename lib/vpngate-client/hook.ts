import csv from 'csvtojson';
import {useCallback, useState} from 'react';
import compactMap from '../common/compactMap';
import RealmContext from '../db/realmContext';
import {VPN_GATE_SERVERS_LIST_URL} from './constants';
import {VpnServerRepository} from './interface';
import {IVpnServer, VpnServer} from './models';

//#region Typedefs

export type FetchVpnServersAction = () => Promise<readonly IVpnServer[]>;

export interface UseVpnGateClientDeps {
  fetchVpnServersAction: FetchVpnServersAction;
}

//#endregion

//#region Fetch and parse VPNGate response

const converter = csv();

function convertToVpnServer(value: any): IVpnServer {
  return {
    hostName: value['#HostName'],
    ipAddress: value['IP'],
    score: parseInt(value['Score']),
    ping: parseInt(value['Ping']),
    speed: parseInt(value['Speed']),
    country: value['CountryLong'],
    countryCode: value['CountryShort'],
    operator: value['Operator'],
    base64EncodedOvpnConfig: value['OpenVPN_ConfigData_Base64'],
  };
}

export const fetchVpnGateServers: FetchVpnServersAction = async () => {
  const response = await fetch(VPN_GATE_SERVERS_LIST_URL);
  const body = await response.text();

  const lines = body.split(/\r?\n/).map(s => s.trim());
  const data = lines.slice(1).join('\n');

  return converter
    .fromString(data)
    .then(values => compactMap(values, convertToVpnServer));
};

//#endregion

//#region Hook

export function useVpnGateClient(
  deps: UseVpnGateClientDeps = {
    fetchVpnServersAction: fetchVpnGateServers,
  },
): VpnServerRepository {
  const {fetchVpnServersAction} = deps;
  const {useRealm, useQuery} = RealmContext;

  const [isFetching, setIsFetching] = useState(() => false);
  const realm = useRealm();
  const result = useQuery(VpnServer);
  const dispatch = useCallback(() => {
    setIsFetching(true);

    fetchVpnServersAction().then(vpnServers => {
      realm.write(() => {
        for (const server of vpnServers) {
          realm.create(VpnServer, VpnServer.generate(server));
        }
      });

      setIsFetching(false);
    });
  }, []);

  return {
    vpnServers: result,
    isFetching,
    fetchServers: dispatch,
  };
}

//#endregion
