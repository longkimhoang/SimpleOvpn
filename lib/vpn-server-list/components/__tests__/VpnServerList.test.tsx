import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {IVpnServer} from '../../../vpngate-client/models';
import VpnServerList, {exportedForTesting} from '../VpnServerList';

describe('VpnServerList', () => {
  function generateVpnServers(count: number): IVpnServer[] {
    let servers: IVpnServer[] = [];

    for (let i = 0; i < count; ++i) {
      const mockVpnServer: IVpnServer = {
        hostName: 'public-vpn-90',
        ipAddress: `219.100.37.55-${i}`,
        score: 3037585,
        ping: 16,
        speed: 71957702,
        country: 'Japan',
        countryCode: 'JP',
        operator: 'Daiyuu Nobori_ Japan. Academic Use Only.',
        base64EncodedOvpnConfig: 'mockBase64OvpnConfig',
      };

      servers.push(mockVpnServer);
    }

    return servers;
  }

  test('render empty state', () => {
    const tree = render(<VpnServerList vpnServers={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('render items', () => {
    const items = generateVpnServers(5);
    const tree = render(<VpnServerList vpnServers={items} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('getVpnServerCellLayout called correctly', () => {
    const {getVpnServerCellLayout} = exportedForTesting;
    const items = generateVpnServers(2);
    const layouts = items.map((_, index, data) => {
      return getVpnServerCellLayout(data, index);
    });

    const expectedLayouts = [
      {
        length: 60,
        offset: 0,
        index: 0,
      },
      {
        length: 60,
        offset: 60,
        index: 1,
      },
    ];

    expect(layouts).toEqual(expectedLayouts);
  });

  test('should fire event handler correctly', () => {
    const items = generateVpnServers(1);
    const mockOnPress = jest.fn();
    const {queryByTestId} = render(
      <VpnServerList vpnServers={items} onItemPress={mockOnPress} />,
    );

    const cellElement = queryByTestId('vpn-server-list.cell');
    fireEvent(cellElement!, 'onPress');

    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(mockOnPress).toHaveBeenCalledWith(items[0]);
  });
});
