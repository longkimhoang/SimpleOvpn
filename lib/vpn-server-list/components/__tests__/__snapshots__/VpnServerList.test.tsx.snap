// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VpnServerList render empty state 1`] = `
<RCTScrollView
  collapsable={false}
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  getItemLayout={[Function]}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  initialNumToRender={8}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(255, 251, 254, 1)",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
  windowSize={5}
>
  <View />
</RCTScrollView>
`;

exports[`VpnServerList render items 1`] = `
<RCTScrollView
  collapsable={false}
  data={
    Array [
      Object {
        "base64EncodedOvpnConfig": "mockBase64OvpnConfig",
        "country": "Japan",
        "countryCode": "JP",
        "hostName": "public-vpn-90",
        "ipAddress": "219.100.37.55-0",
        "operator": "Daiyuu Nobori_ Japan. Academic Use Only.",
        "ping": 16,
        "score": 3037585,
        "speed": 71957702,
      },
      Object {
        "base64EncodedOvpnConfig": "mockBase64OvpnConfig",
        "country": "Japan",
        "countryCode": "JP",
        "hostName": "public-vpn-90",
        "ipAddress": "219.100.37.55-1",
        "operator": "Daiyuu Nobori_ Japan. Academic Use Only.",
        "ping": 16,
        "score": 3037585,
        "speed": 71957702,
      },
      Object {
        "base64EncodedOvpnConfig": "mockBase64OvpnConfig",
        "country": "Japan",
        "countryCode": "JP",
        "hostName": "public-vpn-90",
        "ipAddress": "219.100.37.55-2",
        "operator": "Daiyuu Nobori_ Japan. Academic Use Only.",
        "ping": 16,
        "score": 3037585,
        "speed": 71957702,
      },
      Object {
        "base64EncodedOvpnConfig": "mockBase64OvpnConfig",
        "country": "Japan",
        "countryCode": "JP",
        "hostName": "public-vpn-90",
        "ipAddress": "219.100.37.55-3",
        "operator": "Daiyuu Nobori_ Japan. Academic Use Only.",
        "ping": 16,
        "score": 3037585,
        "speed": 71957702,
      },
      Object {
        "base64EncodedOvpnConfig": "mockBase64OvpnConfig",
        "country": "Japan",
        "countryCode": "JP",
        "hostName": "public-vpn-90",
        "ipAddress": "219.100.37.55-4",
        "operator": "Daiyuu Nobori_ Japan. Academic Use Only.",
        "ping": 16,
        "score": 3037585,
        "speed": 71957702,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  getItemLayout={[Function]}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  initialNumToRender={8}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(255, 251, 254, 1)",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
  windowSize={5}
>
  <View>
    <View
      style={null}
    >
      <RNGestureHandlerButton
        collapsable={false}
        exclusive={true}
        handlerTag={3}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="vpn-server-list.cell"
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            Object {
              "height": 60,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginVertical": 6,
                    "paddingLeft": 8,
                  },
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                    ],
                  ]
                }
              >
                public-vpn-90.opengw.net
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(73, 69, 79, 1)",
                      },
                    ],
                  ]
                }
              >
                219.100.37.55-0
              </Text>
            </View>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
    <View
      style={null}
    >
      <RNGestureHandlerButton
        collapsable={false}
        exclusive={true}
        handlerTag={4}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="vpn-server-list.cell"
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            Object {
              "height": 60,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginVertical": 6,
                    "paddingLeft": 8,
                  },
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                    ],
                  ]
                }
              >
                public-vpn-90.opengw.net
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(73, 69, 79, 1)",
                      },
                    ],
                  ]
                }
              >
                219.100.37.55-1
              </Text>
            </View>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
    <View
      style={null}
    >
      <RNGestureHandlerButton
        collapsable={false}
        exclusive={true}
        handlerTag={5}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="vpn-server-list.cell"
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            Object {
              "height": 60,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginVertical": 6,
                    "paddingLeft": 8,
                  },
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                    ],
                  ]
                }
              >
                public-vpn-90.opengw.net
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(73, 69, 79, 1)",
                      },
                    ],
                  ]
                }
              >
                219.100.37.55-2
              </Text>
            </View>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
    <View
      style={null}
    >
      <RNGestureHandlerButton
        collapsable={false}
        exclusive={true}
        handlerTag={6}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="vpn-server-list.cell"
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            Object {
              "height": 60,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginVertical": 6,
                    "paddingLeft": 8,
                  },
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                    ],
                  ]
                }
              >
                public-vpn-90.opengw.net
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(73, 69, 79, 1)",
                      },
                    ],
                  ]
                }
              >
                219.100.37.55-3
              </Text>
            </View>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
    <View
      style={null}
    >
      <RNGestureHandlerButton
        collapsable={false}
        exclusive={true}
        handlerTag={7}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        testID="vpn-server-list.cell"
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            Object {
              "height": 60,
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginVertical": 6,
                    "paddingLeft": 8,
                  },
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                    ],
                  ]
                }
              >
                public-vpn-90.opengw.net
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(73, 69, 79, 1)",
                      },
                    ],
                  ]
                }
              >
                219.100.37.55-4
              </Text>
            </View>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
</RCTScrollView>
`;
